src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -538,14 +538,15 @@ public void acceptRecord(final SAMRecord record) {
                         boolean isOnTarget = false;
                         for (final Interval target : targets) {
                             if (refPos >= target.getStart() && refPos <= target.getEnd()) {
-                                isOnTarget = true;
                                 ++this.metrics.ON_TARGET_BASES;
                                 if (mappedInPair) ++this.metrics.ON_TARGET_FROM_PAIR_BASES;
 
                                 final int targetOffset = refPos - target.getStart();
                                 final Coverage coverage = this.coverageByTarget.get(target);
                                 coverage.addBase(targetOffset);
                                 baseQHistogramArray[baseQualities[offset]]++;
+                                if (!isOnTarget) coverage.addReadCount();
+                                isOnTarget = true;
```

Comment:
Oh, and will also increment the same target multiple times if there's an indel, causing the read to have multiple alignment blocks overlapping the same target.